common/beerocks/bcl/source/network/network_utils.cpp

```diff
@@ -1030,21 +1030,16 @@ bool network_utils::arp_send(const std::string &iface, const std::string &dst_ip
                              const std::string &src_ip, sMacAddr dst_mac, sMacAddr src_mac,
                              int count, int arp_socket)
 {
+    if (dst_ip.empty() || dst_ip == ZERO_IP_STRING) {
+        LOG(ERROR) << "ip addr is empty or filled with zeros";
+        return false;
+    }
+
     int tx_len;
     arp_hdr arphdr;
     struct sockaddr_ll sock;
     uint8_t packet_buffer[128];
-
-    // If the destination IP is empty, there is no point sending the arp, therefore replace the
-    // with broadcast IP, so all clients will receive it and answer, but since the request is
-    // being sent to a specific mac address, then only the requested client will answer.
```

Comment:
You need to have a better reason than "not working well"  to remove this code.
It used to work so clearly something broke , or maybe it does work on some clients (another reason to keep it) 
Also , this commit is not related to this PR and you cannot close  #1056 until this issue is resolved or at least debugged.